File: app/src/main/java/com/cookiegames/smartcookie/di/AppComponent.java
package com.cookiegames.smartcookie.di;

import javax.inject.Singleton;

import com.cookiegames.smartcookie.adblock.AssetsAdBlocker;
import com.cookiegames.smartcookie.adblock.NoOpAdBlocker;
import com.cookiegames.smartcookie.browser.activity.BrowserActivity;
import com.cookiegames.smartcookie.malwareblock.AssetsMalwareBlock;
import com.cookiegames.smartcookie.malwareblock.NoOpMalwareBlock;
import com.cookiegames.smartcookie.reading.activity.ReadingActivity;
import com.cookiegames.smartcookie.browser.TabsManager;
import com.cookiegames.smartcookie.browser.activity.ThemableBrowserActivity;
import com.cookiegames.smartcookie.settings.activity.ThemableSettingsActivity;
import com.cookiegames.smartcookie.BrowserApp;
import com.cookiegames.smartcookie.browser.BrowserPresenter;
import com.cookiegames.smartcookie.browser.SearchBoxModel;
import com.cookiegames.smartcookie.constant.BookmarkPage;
import com.cookiegames.smartcookie.constant.DownloadsPage;
import com.cookiegames.smartcookie.constant.HistoryPage;
import com.cookiegames.smartcookie.constant.StartPage;
import com.cookiegames.smartcookie.dialog.LightningDialogBuilder;
import com.cookiegames.smartcookie.download.DownloadHandler;
import com.cookiegames.smartcookie.download.LightningDownloadListener;
import com.cookiegames.smartcookie.settings.fragment.BookmarkSettingsFragment;
import com.cookiegames.smartcookie.browser.fragment.BookmarksFragment;
import com.cookiegames.smartcookie.settings.fragment.DebugSettingsFragment;
import com.cookiegames.smartcookie.settings.fragment.GeneralSettingsFragment;
import com.cookiegames.smartcookie.settings.fragment.LightningPreferenceFragment;
import com.cookiegames.smartcookie.settings.fragment.PrivacySettingsFragment;
import com.cookiegames.smartcookie.browser.fragment.TabsFragment;
import com.cookiegames.smartcookie.search.SearchEngineProvider;
import com.cookiegames.smartcookie.search.SuggestionsAdapter;
import com.cookiegames.smartcookie.utils.ProxyUtils;
import com.cookiegames.smartcookie.view.LightningChromeClient;
import com.cookiegames.smartcookie.view.LightningView;
import com.cookiegames.smartcookie.view.LightningWebClient;
import dagger.Component;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(BrowserActivity activity);

    void inject(BookmarksFragment fragment);

    void inject(BookmarkSettingsFragment fragment);

    void inject(LightningDialogBuilder builder);

    void inject(TabsFragment fragment);

    void inject(LightningView lightningView);

    void inject(ThemableBrowserActivity activity);

    void inject(LightningPreferenceFragment fragment);

    void inject(BrowserApp app);

    void inject(ProxyUtils proxyUtils);

    void inject(ReadingActivity activity);

    void inject(LightningWebClient webClient);

    void inject(ThemableSettingsActivity activity);

    void inject(LightningDownloadListener listener);

    void inject(PrivacySettingsFragment fragment);

    void inject(StartPage startPage);

    void inject(HistoryPage historyPage);

    void inject(BookmarkPage bookmarkPage);

    void inject(DownloadsPage downloadsPage);

    void inject(BrowserPresenter presenter);

    void inject(TabsManager manager);

    void inject(DebugSettingsFragment fragment);

    void inject(SuggestionsAdapter suggestionsAdapter);

    void inject(LightningChromeClient chromeClient);

    void inject(DownloadHandler downloadHandler);

    void inject(SearchBoxModel searchBoxModel);

    void inject(SearchEngineProvider searchEngineProvider);

    void inject(GeneralSettingsFragment generalSettingsFragment);

    //void inject(NetworkObservable networkObservable);

    AssetsAdBlocker provideAssetsAdBlocker();

    NoOpAdBlocker provideNoOpAdBlocker();

    AssetsMalwareBlock provideAssetsMalwareBlock();

    NoOpMalwareBlock provideNoOpMalwareBlock();

}

File: app/src/main/java/com/cookiegames/smartcookie/di/AppComponent.kt
package com.cookiegames.smartcookie.di

import com.cookiegames.smartcookie.BrowserApp
import com.cookiegames.smartcookie.adblock.BloomFilterAdBlocker
import com.cookiegames.smartcookie.adblock.NoOpAdBlocker
import com.cookiegames.smartcookie.browser.SearchBoxModel
import com.cookiegames.smartcookie.browser.activity.BrowserActivity
import com.cookiegames.smartcookie.browser.activity.ThemableBrowserActivity
import com.cookiegames.smartcookie.browser.bookmarks.BookmarksDrawerView
import com.cookiegames.smartcookie.device.BuildInfo
import com.cookiegames.smartcookie.dialog.LightningDialogBuilder
import com.cookiegames.smartcookie.download.LightningDownloadListener
import com.cookiegames.smartcookie.reading.activity.ReadingActivity
import com.cookiegames.smartcookie.search.SuggestionsAdapter
import com.cookiegames.smartcookie.settings.activity.SettingsActivity
import com.cookiegames.smartcookie.settings.activity.ThemableSettingsActivity
import com.cookiegames.smartcookie.settings.fragment.*
import com.cookiegames.smartcookie.view.SmartCookieChromeClient
import com.cookiegames.smartcookie.view.SmartCookieView
import com.cookiegames.smartcookie.view.SmartCookieWebClient
import android.app.Application
import com.cookiegames.smartcookie.download.DownloadActivity
import com.cookiegames.smartcookie.history.HistoryActivity
import com.cookiegames.smartcookie.popup.PopUpClass
import dagger.BindsInstance
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [(AppModule::class), (AppBindsModule::class)])
interface AppComponent {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        @BindsInstance
        fun buildInfo(buildInfo: BuildInfo): Builder

        fun build(): AppComponent
    }

    fun inject(activity: BrowserActivity)

    fun inject(activity: DownloadActivity)

    fun inject(activity: HistoryActivity)

    fun inject(fragment: ExportSettingsFragment)

    fun inject(builder: LightningDialogBuilder)

    fun inject(smartCookieView: SmartCookieView)

    fun inject(activity: ThemableBrowserActivity)

    fun inject(advancedSettingsFragment: AdvancedSettingsFragment)

    fun inject(app: BrowserApp)

    fun inject(activity: ReadingActivity)

    fun inject(webClient: SmartCookieWebClient)

    fun inject(activity: SettingsActivity)

    fun inject(activity: ThemableSettingsActivity)

    fun inject(listener: LightningDownloadListener)

    fun inject(fragment: PrivacySettingsFragment)

    fun inject(fragment: DebugSettingsFragment)

    fun inject(fragment: ExtensionsSettingsFragment)

    fun inject(suggestionsAdapter: SuggestionsAdapter)

    fun inject(chromeClient: SmartCookieChromeClient)

    fun inject(searchBoxModel: SearchBoxModel)

    fun inject(generalSettingsFragment: GeneralSettingsFragment)

    fun inject(displaySettingsFragment: DisplaySettingsFragment)

    fun inject(adBlockSettingsFragment: AdBlockSettingsFragment)

    fun inject(drawerSettingsFragment: DrawerSettingsFragment)

    fun inject(homepageSettingsFragment: HomepageSettingsFragment)

    fun inject(themeSettingsFragment: ThemeSettingsFragment)

    fun inject(drawerOffsetFragment: DrawerOffsetFragment)

    fun inject(parentalSettingsFragment: ParentalControlSettingsFragment)

    fun inject(bookmarksView: BookmarksDrawerView)

    fun provideBloomFilterAdBlocker(): BloomFilterAdBlocker

    fun provideNoOpAdBlocker(): NoOpAdBlocker

    fun inject(popUpClass: PopUpClass)
}
